ANNOTATION.md

```diff
@@ -1,142 +1,142 @@
+| /self::\*\[@InternalType='CompilationUnit'\]//\*\[@InternalType='PrefixExpression'\]/self::\*\[@operator\]\[@operator='\+\+'\] | Increment |
+| /self::\*\[@InternalType='CompilationUnit'\]//\*\[@InternalType='PrefixExpression'\]/self::\*\[@operator\]\[@operator='\-\-'\] | Decrement |
+| /self::\*\[@InternalType='CompilationUnit'\]//\*\[@InternalType='PrefixExpression'\]/self::\*\[@operator\]\[@operator='\+'\] | Positive |
+| /self::\*\[@InternalType='CompilationUnit'\]//\*\[@InternalType='PrefixExpression'\]/self::\*\[@operator\]\[@operator='\-'\] | Negative |
+| /self::\*\[@InternalType='CompilationUnit'\]//\*\[@InternalType='PrefixExpression'\]/self::\*\[@operator\]\[@operator='~'\] | Bitwise, Not |
+| /self::\*\[@InternalType='CompilationUnit'\]//\*\[@InternalType='PrefixExpression'\]/self::\*\[@operator\]\[@operator='\!'\] | Boolean, Not |
+| /self::\*\[@InternalType='CompilationUnit'\]//\*\[@InternalType='Assignment'\] | Expression, Assignment, Operator, Binary |
+| /self::\*\[@InternalType='CompilationUnit'\]//\*\[@InternalType='Assignment'\]/\*\[@internalRole\]\[@internalRole='leftHandSide'\] | Assignment, Binary, Left |
+| /self::\*\[@InternalType='CompilationUnit'\]//\*\[@InternalType='Assignment'\]/\*\[@internalRole\]\[@internalRole='rightHandSide'\] | Assignment, Binary, Right |
+| /self::\*\[@InternalType='CompilationUnit'\]//\*\[@InternalType='Assignment'\]/self::\*\[@operator\]\[@operator='\+='\] | Add |
+| /self::\*\[@InternalType='CompilationUnit'\]//\*\[@InternalType='Assignment'\]/self::\*\[@operator\]\[@operator='\-='\] | Substract |
+| /self::\*\[@InternalType='CompilationUnit'\]//\*\[@InternalType='Assignment'\]/self::\*\[@operator\]\[@operator='\*='\] | Multiply |
+| /self::\*\[@InternalType='CompilationUnit'\]//\*\[@InternalType='Assignment'\]/self::\*\[@operator\]\[@operator='/='\] | Divide |
+| /self::\*\[@InternalType='CompilationUnit'\]//\*\[@InternalType='Assignment'\]/self::\*\[@operator\]\[@operator='%='\] | Modulo |
+| /self::\*\[@InternalType='CompilationUnit'\]//\*\[@InternalType='Assignment'\]/self::\*\[@operator\]\[@operator='&='\] | Bitwise, And |
+| /self::\*\[@InternalType='CompilationUnit'\]//\*\[@InternalType='Assignment'\]/self::\*\[@operator\]\[@operator='\|='\] | Bitwise, Or |
+| /self::\*\[@InternalType='CompilationUnit'\]//\*\[@InternalType='Assignment'\]/self::\*\[@operator\]\[@operator='^='\] | Bitwise, Xor |
+| /self::\*\[@InternalType='CompilationUnit'\]//\*\[@InternalType='Assignment'\]/self::\*\[@operator\]\[@operator='<<='\] | Bitwise, LeftShift |
+| /self::\*\[@InternalType='CompilationUnit'\]//\*\[@InternalType='Assignment'\]/self::\*\[@operator\]\[@operator='>>='\] | Bitwise, RightShift |
+| /self::\*\[@InternalType='CompilationUnit'\]//\*\[@InternalType='Assignment'\]/self::\*\[@operator\]\[@operator='>>>='\] | Bitwise, RightShift, Unsigned |
+| /self::\*\[@InternalType='CompilationUnit'\]//\*\[@InternalType='ArrayType'\] | Type, Primitive, List |
```

Comment:
I'm not sure `Primitive`+`List` is the same as an array. We probably need an Array role for consecutive access memory.

Reply:
Yes, I agree. I used `List` because `List` role is currently defined as a generic sequence (as it was in SDKv0)